src/ragQueries.ts

```diff
@@ -0,0 +1,42 @@
+import { GraphAnnotation } from "./agent/agent"
+import { OpenAIEmbeddings } from "@langchain/openai";
+import { Chroma } from "@langchain/community/vectorstores/chroma";
+
+const embeddings = new OpenAIEmbeddings({
+    model: "text-embedding-3-small",
+    apiKey: process.env.OPENAI_API_KEY || ""
+});
+
+const vectorStore = new Chroma(embeddings, {
+    collectionName: "uw-courses-test",
+    url: "http://localhost:8080", // Optional, will default to this value
+    collectionMetadata: {
+      "hnsw:space": "cosine",
+    }, // Optional, can be used to specify the distance method of the embedding space https://docs.trychroma.com/usage-guide#changing-the-distance-function
+  });
+
+
+// Thoughts: can probably turn this into a retreiver: https://js.langchain.com/docs/integrations/vectorstores/chroma/#query-by-turning-into-retriever
+// vector similarity search 
+export const vectorSimilaritySearch = async (state: typeof GraphAnnotation.State) => {
+    console.log("vectorSimilaritySearch");
+
+    // perform vector similarity search
+    // vectorStore.similaritySearchWithScore can be used to get scores if we want to filter by score
+    const documents = await vectorStore.similaritySearch(state.semanticSearchQuery || state.rawUserChat, 3);
+    
+    // check if there are documents
+    if (documents.length === 0) {
+        console.log('No matching documents found.');
+        return { context: '', documents: [] };
+    }
+
+    // combine the context from matching documents
+    const context = documents.map((doc) => doc.pageContent).join('\n\n - -\n\n');
```

Comment:
Joining map: https://medium.com/@callumjmac/implementing-rag-in-langchain-with-chroma-a-step-by-step-guide-16fc21815339

Need to test

Reply:
If we have both a context string and documents array (and the context string can be formed from the documents) we should either just return the documents array (which would be the retrieved info formatted into json objects) or return the context string. Im in favor of returning the documents array, and then in the answering prompt we can write code to convert the document string into something that can be injected into the prompt.